src/main/scala/au/com/agiledigital/toolform/tasks/generate/docker/DockerFormatting.scala

```diff
@@ -1,32 +1,19 @@
 package au.com.agiledigital.toolform.tasks.generate.docker
 
-import au.com.agiledigital.toolform.model.{Component, SubEdgeType}
+import au.com.agiledigital.toolform.model.{PortMapping, SubEdgeType}
+import au.com.agiledigital.toolform.tasks.generate.Formatting.normaliseImageName
 
 /**
-  * A collection of pure formatting functions for use by the GenerateDockerComposeV3 class.
+  * A collection of pure formatting functions for use for Docker Compose file generation.
   */
 object DockerFormatting {
-  def normaliseServiceId(name: String): String =
-    name
-      .replace("/", "-")
-      .replace(" ", "-")
-      .replace("_", "-")
-      .toLowerCase
 
-  def normaliseImageName(name: String): String =
-    name
-      .replace(" ", "_")
-      .replace("-", "_")
-      .toLowerCase
-
-  def componentServiceName(component: Component): String =
-    normaliseServiceId(component.id)
-
-  def subEdgeServiceName(projectId: String, subEdgeDef: SubEdgeDef): String =
-    normaliseServiceId(s"$projectId-${subEdgeDef.edgeId}-${subEdgeDef.subEdgeId}-nginx")
+  def formatEnvironment(entry: (String, String)): String = s"- ${entry._1}=${entry._2}"
 
-  def componentImageName(projectId: String, component: Component): String =
-    normaliseImageName(s"$projectId/${component.id}")
+  def formatPort(port: String): String = {
+    val normalisedPort = PortMapping.parsePortMappingFromConfigString(port).toString
```

Comment:
This should be done when the project is read.

Reply:
Fixed in c2dd744